docs/rules/prefer-function-component.md

```diff
@@ -0,0 +1,89 @@
+# Prefer function components over class components (react/prefer-function-component)
+
+This rule prevents the use of class components.
+
+Since the addition of hooks, it has been possible to write stateful React components
+using only functions. Mixing both class and function components in a code base adds unnecessary hurdles for sharing reusable logic.
+
+By default, class components that use `componentDidCatch` are enabled because there is currently no hook alternative for React. This option is configurable via `allowComponentDidCatch`.
```

Comment:
i'm not clear on why this is an option. A class component that's an error boundary *can never* be a functional component (pending changes from React itself), so there'd never be any value in this rule warning on it. I think that this option shouldn't be an option, it should just be hardcoded to true.

Reply:
Many codebases I've seen don't implement error boundaries from scratch in their code base, instead they import library solutions that use error boundaries internally. I made this configurable (though with the default value of `true` so most clients can ignore it) for clients who want _all_ class component usage to be flagged. This also provides a migration path should React release functional component error boundaries.

Reply:
Also, Preact does offer functional error boundaries via a hook, so this would be valuable for those users.